iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/AreaImpl.java

```diff
@@ -0,0 +1,282 @@
+            } else if (identifiable instanceof Connectable<?> connectable) {
+                // if connectable removed from network, remove its terminals from this extension
+                connectable.getTerminals().forEach(AreaImpl.this::removeAreaBoundary);
+            }
+            // When a VoltageLevel is removed, its areas' voltageLevels attributes are directly updated. This is not managed via this listener.
+        }
+    }
+
+    private final NetworkListener areaListener;
+
+    public AreaImpl(Ref<NetworkImpl> ref, Ref<SubnetworkImpl> subnetworkRef, String id, String name, boolean fictitious, String areaType,
+                    double acInterchangeTarget) {
+        super(id, name, fictitious);
+        this.networkRef = Objects.requireNonNull(ref);
+        this.subnetworkRef = subnetworkRef;
+        this.areaType = Objects.requireNonNull(areaType);
+        this.voltageLevels = new LinkedHashSet<>();
+        this.areaBoundaries = new ArrayList<>();
+
+        int variantArraySize = networkRef.get().getVariantManager().getVariantArraySize();
+        this.acInterchangeTarget = new TDoubleArrayList(variantArraySize);
+        for (int i = 0; i < variantArraySize; i++) {
+            this.acInterchangeTarget.add(acInterchangeTarget);
+        }
+        this.areaListener = new AreaListener();
+        getNetwork().addListener(this.areaListener);
+    }
+
+    @Override
+    public NetworkImpl getNetwork() {
+        throwIfRemoved("network");
+        return networkRef.get();
+    }
+
+    @Override
+    public Network getParentNetwork() {
+        throwIfRemoved("network");
+        return Optional.ofNullable((Network) subnetworkRef.get()).orElse(getNetwork());
+    }
+
+    @Override
+    protected String getTypeDescription() {
+        return "Area";
+    }
+
+    @Override
+    public String getAreaType() {
+        throwIfRemoved("area type");
+        return areaType;
+    }
+
+    @Override
+    public Iterable<VoltageLevel> getVoltageLevels() {
+        throwIfRemoved("voltage levels");
+        return voltageLevels;
+    }
+
+    @Override
+    public Stream<VoltageLevel> getVoltageLevelStream() {
+        throwIfRemoved("voltage levels");
+        return voltageLevels.stream();
+    }
+
+    @Override
+    public OptionalDouble getAcInterchangeTarget() {
+        throwIfRemoved("AC interchange target");
+        double target = acInterchangeTarget.get(getNetwork().getVariantIndex());
+        if (Double.isNaN(target)) {
+            return OptionalDouble.empty();
+        }
+        return OptionalDouble.of(target);
+    }
+
+    @Override
+    public Area setAcInterchangeTarget(double acInterchangeTarget) {
+        NetworkImpl n = getNetwork();
+        int variantIndex = n.getVariantIndex();
+        double oldValue = this.acInterchangeTarget.set(variantIndex, acInterchangeTarget);
+        String variantId = n.getVariantManager().getVariantId(variantIndex);
+        n.invalidateValidationLevel();
```

Comment:
There is no need to invalidate the validation level because there is no check about the AC interchange target regarding the validation level (see `ValidationUtil.validate(…)`.

Reply:
thanks, I wasn't fully aware on how this feature works, indeed doesn't apply here, this is fixed now.